### PR TITLE
Deprecate GCC support for compiling Open Enclave SDK for Intel SGX from source

### PR DESCRIPTION
**Pull request by CyanDevs**
Signed-off-by: Chris Yan <chrisyan@microsoft.com>

### CODE REVIEW

**Comment by BRMcLaren** [comment]
bors try

**Comment by BRMcLaren** [comment]
This code does remove all the GCC components from the testing framework, but I believe we need to continue to test the packages themselves with GCC. Is it possible that doing something along the lines of refactoring this section of code would help accomplish this? https://github.com/openenclave/openenclave-ci/blob/master/src/jenkins/common/Openenclave.groovy#L33

File: .jenkins/pipelines/Agnostic/Linux/Jenkinsfile [hunk 1]
```diff
@@ -38,27 +38,6 @@ def simulationContainerTest(String version, String build_type, List extra_cmake_
     }
 }
 
-def AArch64GNUTest(String version, String build_type) {
```
Comment (mingweishih) [inline .jenkins/pipelines/Agnostic/Linux/Jenkinsfile:41]: We need to keep this.

File: .jenkins/pipelines/Agnostic/Linux/Jenkinsfile [hunk 2]
```diff
@@ -102,10 +81,6 @@ try{
         "Check CI":                                { checkCI() },
         "Check Developer Experience Ubuntu 16.04": { checkDevFlows('16.04') },
         "Check Developer Experience Ubuntu 18.04": { checkDevFlows('18.04') },
-        "AArch64 1604 GNU gcc Debug":              { AArch64GNUTest('16.04', 'Debug') },
-        "AArch64 1604 GNU gcc Release":            { AArch64GNUTest('16.04', 'Release') },
-        "AArch64 1804 GNU gcc Debug":              { AArch64GNUTest('18.04', 'Debug') },
```
Comment (mingweishih) [inline .jenkins/pipelines/Agnostic/Linux/Jenkinsfile:107]: We should retain the aarch64 configurations. 1604 ones can be dropped later along with deprecating 1604 support.

**Review by mingweishih — CHANGES_REQUESTED** [review]

File: .jenkins/pipelines/Agnostic/Linux/Jenkinsfile (hunk 2 above)
Comment (BRMcLaren) [reply .jenkins/pipelines/Agnostic/Linux/Jenkinsfile:107]: Should we maybe rename the 'gcc' part to not mislead?

I took a look at the code [here](https://github.com/openenclave/openenclave-ci/blob/master/src/jenkins/common/Openenclave.groovy#L66) which the tests rely on and it seems it doesn't have anything to do with gcc?

Comment (mingweishih) [reply .jenkins/pipelines/Agnostic/Linux/Jenkinsfile:107]: It looks like
https://github.com/openenclave/openenclave/blob/3b6f413686846ec3b47ef7e94824c4078c2546f6/.jenkins/pipelines/Agnostic/Linux/Jenkinsfile#L51
will default to aarch gcc tool chain.

Comment (BRMcLaren) [reply .jenkins/pipelines/Agnostic/Linux/Jenkinsfile:107]: I think your understanding of this matter may far exceed ours. If we are to deprecate gcc, that to me sounds like then this needs to be removed. Should the process instead be to deprecate gcc from docs and ci but keep these tests.

I looked into it a little, but it seems to me that GNU GCC != GCC which would align with what you are saying, whereas GNU GCC is for cross compilaiton.

Comment (CyanDevs) [reply .jenkins/pipelines/Agnostic/Linux/Jenkinsfile]: As I understand it, the reasoning is to keep GCC support for OPTEE (AArch64) even though we are deprecating it for SGX. I have pushed new changes that keeps this.

**Comment by BRMcLaren** [comment]
bors try

**Review by radhikaj — APPROVED** [review]

**Review by mingweishih — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 [Unreleased][Unreleased_log]
 --------------
+### Deprecated
+
+- Compiling Open Enclave SDK from source with GCC is no longer supported. The recommended compiler is Clang.
```
Comment (mingweishih) [inline CHANGELOG.md:14]:
```suggestion
- Compiling Open Enclave SDK for Intel SGX from source with GCC is no longer supported. The recommended compiler is Clang.
```

File: docs/GettingStartedDocs/buildandsign.md [hunk 4]
```diff
@@ -8,15 +8,15 @@ Also like the Intel SGX SDK, these binaries must be built with a specific set of
 To simplify the process of the specifying the correct build parameters, Open Enclave SDK provides
 a set of `pkg-config` settings that you can use in your build scripts.
 
-There are 8 configurations provided to handle all permutations of the following parameters:
```
Comment (mingweishih) [inline docs/GettingStartedDocs/buildandsign.md:11]: This document is related to building samples and does not need to modify.

File: CHANGELOG.md (hunk 3 above)
Comment (CyanDevs) [reply CHANGELOG.md]: Good suggestion for clarification. Thank you!

File: docs/GettingStartedDocs/buildandsign.md (hunk 4 above)
Comment (CyanDevs) [reply docs/GettingStartedDocs/buildandsign.md]: Thanks for the clarification, I have reverted the changes to this file.

File: docs/GettingStartedDocs/buildandsign.md [hunk 5]
```diff
@@ -97,40 +97,9 @@ NumHeapPages=1024
 NumStackPages=1024
 NumTCS=1
 ```
-Additionally, a developer can specify additional Key Sharing and Separation (KSS) identity properties
```
Comment (BRMcLaren) [inline docs/GettingStartedDocs/buildandsign.md:100]: Is this GCC specific? trying to understand the deletion.

Comment (CyanDevs) [reply docs/GettingStartedDocs/buildandsign.md]: Good catch, this shouldn't have been changed.

**Pull request merged**